pallets/connectors-gateway/src/tests.rs

```diff
@@ -0,0 +1,480 @@
+use cfg_mocks::*;
+use cfg_traits::connectors::{Codec, OutboundQueue};
+use cfg_types::domain_address::*;
+use frame_support::{assert_noop, assert_ok};
+use sp_core::{crypto::AccountId32, ByteArray, H160};
+use sp_runtime::DispatchError::BadOrigin;
+
+use super::{
+	mock::{RuntimeEvent as MockEvent, *},
+	origin::*,
+	pallet::*,
+};
+
+mod utils {
+	use super::*;
+
+	pub fn get_random_test_account_id() -> AccountId32 {
+		rand::random::<[u8; 32]>().into()
+	}
+
+	pub fn event_exists<E: Into<MockEvent>>(e: E) {
+		let actual: Vec<MockEvent> = frame_system::Pallet::<Runtime>::events()
+			.iter()
+			.map(|e| e.event.clone())
+			.collect();
+
+		let e: MockEvent = e.into();
+		let mut exists = false;
+		for evt in actual {
+			if evt == e {
+				exists = true;
+				break;
+			}
+		}
```

Comment:
Nit: Could be simplified
```suggestion
		let e: MockEvent = e.into();
		let exists = actual.contains(&e);
```

Reply:
Or the entire function:
```rust
pub fn event_exists<E: Into<MockEvent>>(e: E) {
    let e: MockEvent = e.into();
    assert!(frame_system::Pallet::<Runtime>::events().iter().any(|ev| ev.event == e));
}
```